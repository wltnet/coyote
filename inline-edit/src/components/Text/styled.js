import styled from "styled-components";

export const Wrapper = styled.div`
  display: flex;
  justify-content: space-between;
`;

export const InputText = styled.div`
`;

export const Error = styled.div`
  font-size: 0.8rem;
  color: red;
`;
