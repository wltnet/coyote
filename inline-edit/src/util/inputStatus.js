export const INPUT_STATUS = {
  SAVED: "Saved",
  LOADING: "Loading",
  SUCCESS: "Success",
  ERROR: "Error",
}
