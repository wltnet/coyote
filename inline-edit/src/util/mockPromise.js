export const saveTextSuccess = (delay, value) => new Promise((resolve, reject) => {
  // 50:50 chance for resolve and reject
  const random_result = Math.random() >= 0.5 ? resolve : reject;
  setTimeout(random_result, delay, value);
});
