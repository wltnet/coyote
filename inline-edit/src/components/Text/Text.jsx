import React from "react";
import Icon from "../Icon";
import { INPUT_STATUS } from "../../util/inputStatus";
import { Error, Wrapper, InputText } from "./styled";

const Text = ({ text, handleClick, status }) => (
<>
  <Wrapper onClick={handleClick}>
    <InputText>{text}</InputText>
    <Icon status={status} />
  </Wrapper>
  {status === INPUT_STATUS.ERROR && <Error>Oops! Something has gone terribly wrong!</Error> }
</>
);
export default Text;
