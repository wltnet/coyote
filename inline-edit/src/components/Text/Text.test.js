import React from "react";
import { shallow } from "enzyme";
import Text from "./Text";
import { INPUT_STATUS } from "../../util/inputStatus";

jest.mock("./styled", () => ({
  Wrapper: "Wrapper",
  InputText: "InputText",
  Error: "Error",
}));

describe("<Text />", () => {
  it("should match the snapshot when first render", () => {
    const props = {
      text: "Hello World",
      handleClick: jest.fn(),
      status: undefined,
    }
    const wrapper = shallow(<Text {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should match the snapshot when has error status", () => {
    const props = {
      text: "Hello World",
      handleClick: jest.fn(),
      status: INPUT_STATUS.ERROR,
    }
    const wrapper = shallow(<Text {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should trigger handleclick function when click on the text", () => {
    const props = {
      text: "Hello World",
      handleClick: jest.fn(),
      status: undefined,
    }
    const wrapper = shallow(<Text {...props} />);
    wrapper.find("Wrapper").simulate("click");
    expect(props.handleClick).toHaveBeenCalled();
  });
});
