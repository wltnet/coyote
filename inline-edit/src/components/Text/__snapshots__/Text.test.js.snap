// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should match the snapshot when first render 1`] = `
<Fragment>
  <Wrapper
    onClick={[MockFunction]}
  >
    <InputText>
      Hello World
    </InputText>
    <Icon />
  </Wrapper>
</Fragment>
`;

exports[`<Text /> should match the snapshot when has error status 1`] = `
<Fragment>
  <Wrapper
    onClick={[MockFunction]}
  >
    <InputText>
      Hello World
    </InputText>
    <Icon
      status="Error"
    />
  </Wrapper>
  <Error>
    Oops! Something has gone terribly wrong!
  </Error>
</Fragment>
`;
